horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/services/TestService.java

```diff
@@ -87,15 +85,24 @@ TestQueryResult list(@QueryParam("roles") String roles,
                    @QueryParam("sort") @DefaultValue("name") String sort,
                    @QueryParam("direction") SortDirection direction);
 
+   public static final String DEFAULT_LIMIT = "20";
+   public static final String DEFAULT_PAGE = "1";
+
    @Path("summary")
    @GET
    @Operation(description="Retrieve a summary of Tests in a folder")
    @Parameters(value = {
            @Parameter(name = "roles", description = "\"__my\", \"__all\" or a comma delimited  list of roles", example = "__my"),
            @Parameter(name = "folder", description = "name of the Folder containing the Tests", example = "My Team Folder"),
+           @Parameter(name = "limit", description = "limit the result count", example = DEFAULT_LIMIT),
+           @Parameter(name = "page", description = "filter by page number of a paginated list of ", example = DEFAULT_PAGE),
+           @Parameter(name = "direction", description = "Sort direction", example ="Ascending")
    }
    )
-   TestListing summary(@QueryParam("roles") String roles, @QueryParam("folder") String folder);
+   TestListing summary(@QueryParam("roles") String roles, @QueryParam("folder") String folder,
+                  @DefaultValue(DEFAULT_LIMIT) @QueryParam("limit") Integer limit,
+                  @DefaultValue(DEFAULT_PAGE) @QueryParam("page") Integer page,
+                  @DefaultValue("Ascending") @QueryParam("direction") SortDirection direction);
 
```

Comment:
If we are adding defaults to the API, we need to update the openapi spec to reflect this, e.g.;

`@Parameter(name = "limit", description = "limit the result count", example = DEFAULT_LIMIT, schema = @Schema(type = SchemaType.INTEGER , defaultValue = "20") )`

Reply:
thanks for highlighting this is necessary